judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/query/RdbmsBuilder.java

```diff
@@ -488,6 +456,75 @@ private List<RdbmsJoin> processContainerJoin(final ContainerJoin join, final EMa
         return result;
     }
 
+    private List<RdbmsJoin> processCastJoin(Node join, SubSelect parentIdFilterQuery, RdbmsBuilder<ID> rdbmsBuilder, Map<String, Object> queryParameters) {
+        EClass castTargetType = join.getType();
+        Set<EClass> typeSet = new HashSet<>(castTargetType.getEAllSuperTypes());
+        typeSet.add(castTargetType);
+
+        List<EClass> types = typeSet.stream()
+                                    .sorted((l, r) -> {
+                                        // ascending order
+                                        // type < supertype
+                                        if (l.getEAllSuperTypes().contains(r)) {
+                                            return -1;
+                                        }
+                                        if (r.getEAllSuperTypes().contains(l)) {
+                                            return 1;
+                                        }
+                                        return 0;
+                                    })
+                                    .collect(Collectors.toUnmodifiableList());
+
+        List<RdbmsJoin> rdbmsTableJoins = new ArrayList<>();
+        for (EClass type : types) {
+            String alias = join.getAlias();
+            Node partnerTable = ((CastJoin) join).getPartner();
```

Comment:
`String alias = join.getAlias();
 Node partnerTable = ((CastJoin) join).getPartner();`
Put before the for loop.